controllers/categoriesController.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable prettier/prettier */
+const db = require("../models/category");
+
+// Defining methods for the categoriesController
+module.exports = {
+    findAll: (req, res) => {
+        db.Category.find(req.query)
+            .sort({ date: -1 })
+            .then(dbModel => res.json(dbModel))
+            .catch(err => res.status(422).json(err));
+    },
+    findByCategory: (req, res) => {
+        db.Category.findByCategory(req.params.category)
+            .then(dbModel => res.json(dbModel))
+            .catch(err => res.status(422).json(err));
+    },
+    findById: (req, res) => {
+        db.Category.findById(req.params.id)
+            .then(dbModel => res.json(dbModel))
+            .catch(err => res.status(422).json(err));
+    },
+    create: (req, res) => {
+        db.Category.create(req.body)
+            .then(dbModel => res.json(dbModel))
+            .catch(err => res.status(422).json(err));
+    },
+    update: (req, res) => {
+        db.Category.findOneAndUpdate({ _id: req.params.id }, req.body)
```

Comment:
Currently sends the old note (before updates occurred) back to the client. Suggest
```suggestion
        db.Category.findOneAndUpdate({ _id: req.params.id }, req.body, { new: true })
```
Adding the `new` option will send back the updated document